VMEncryption/main/test/test_disk_util.py

```diff
@@ -416,3 +416,115 @@ def test_get_luks_header_size_luks2_badoffset(self, ver_mock, dump_mock):
         dump_mock.return_value = ""
         header_size = self.disk_util.get_luks_header_size("/mocked/device/path")
         self.assertEqual(header_size, None)
+
+    @mock.patch("CommandExecutor.CommandExecutor.Execute", return_value=0)
+    def test_import_token_data(self,cmd_exc_mock):
+        '''update dict type data to luks2 header'''
+        cmd_exc_mock.return_value = 0
+        data = {}
+        result = self.disk_util.import_token_data(device_path="/dev/sda",token_data=data,token_id=1)
+        self.assertEqual(result,False)
+        data = None
+        result = self.disk_util.import_token_data(device_path="/dev/sda",token_data=data,token_id=1)
+        self.assertEqual(result,False)
+        data = [1,3]
+        result = self.disk_util.import_token_data(device_path="/dev/sda",token_data=data,token_id=1)
+        self.assertEqual(result,False)
+        data = "test data"
+        result = self.disk_util.import_token_data(device_path="/dev/sda",token_data=data,token_id=1)
+        self.assertEqual(result,False)
+        data = {'version':'1.0','type':'ADE'}
+        result = self.disk_util.import_token_data(device_path="/dev/sda",token_data=data,token_id=1)
+        self.assertEqual(result,True)
+
+    @mock.patch("os.path.exists", return_value=True)
+    @mock.patch("CommandExecutor.CommandExecutor.Execute", return_value=0)
+    def test_import_token(self,cmd_exc_mock,path_exit):
+        '''read passphrase from Passphrase_file, and update to LUKS2 header'''
+        cmd_exc_mock.return_value = 0
+        path_exit.return_value=False
+        result = self.disk_util.import_token(device_path="/dev/sdc",
+                                             passphrase_file=None,
+                                             public_settings=None)
+        self.assertEqual(result,False)
+        path_exit.return_value=False
+        result = self.disk_util.import_token(device_path="/dev/sdc",
+                                             passphrase_file="/var/lib/file_path",
+                                             public_settings=None)
+        self.assertEqual(result,False)
```

Comment:
Could you add some success cases too if possible?

Reply:
not possible to write success cases.